js/modules/k6/grpc/client_test.go

```diff
@@ -761,3 +766,101 @@ func TestDebugStat(t *testing.T) {
 		})
 	}
 }
+
+func TestResolveFileDescriptors(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name                string
+		pkgs                []string
+		services            []string
+		expectedDescriptors int
+	}{
+		{
+			name:                "SuccessSamePackage",
+			pkgs:                []string{"mypkg"},
+			services:            []string{"Service1", "Service2", "Service3"},
+			expectedDescriptors: 3,
+		},
+		{
+			name:                "SuccessMultiPackages",
+			pkgs:                []string{"mypkg1", "mypkg2", "mypkg3"},
+			services:            []string{"Service", "Service", "Service"},
+			expectedDescriptors: 3,
+		},
+		{
+			name:                "DeduplicateServices",
+			pkgs:                []string{"mypkg1"},
+			services:            []string{"Service1", "Service2", "Service1"},
+			expectedDescriptors: 2,
+		},
+		{
+			name:                "NoServices",
+			services:            []string{},
+			expectedDescriptors: 0,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			var (
+				lsr  = &reflectpb.ListServiceResponse{}
+				mock = &getServiceFileDescriptorMock{}
+			)
+			for i, service := range tt.services {
+				// if only one package is defined then
+				// the package is the same for every service
+				pkg := tt.pkgs[0]
+				if len(tt.pkgs) > 1 {
+					pkg = tt.pkgs[i]
+				}
+
+				lsr.Service = append(lsr.Service, &reflectpb.ServiceResponse{
+					Name: fmt.Sprintf("%s.%s", pkg, service),
+				})
+				mock.pkgs = append(mock.pkgs, pkg)
+				mock.names = append(mock.names, service)
+			}
+
+			fdset, err := resolveServiceFileDescriptors(mock, lsr)
+			require.NoError(t, err)
+			assert.Len(t, fdset.File, tt.expectedDescriptors)
+		})
+	}
+}
+
+type getServiceFileDescriptorMock struct {
+	nreqs int64
+	pkgs  []string
+	names []string
+}
+
+func (m *getServiceFileDescriptorMock) Send(req *reflectpb.ServerReflectionRequest) error {
+	return nil
```

Comment:
Maybe add a TODO to actually check what is sent if it has anything interesting to be tested around it